src/Agent.Worker/TaskManager.cs

```diff
@@ -410,6 +411,68 @@ public sealed class Definition
         public DefinitionData Data { get; set; }
         public string Directory { get; set; }
         public string ZipPath { get; set; }
+
+        public TaskVersion GetPowerShellSDKVersion()
+        {
+            var modulePath = Path.Combine(Directory, "ps_modules", "VstsTaskSdk", "VstsTaskSdk.psd1");
+            if (!File.Exists(modulePath))
+            {
+                return null;
+            }
+
+            var versionLine = File.ReadAllLines(modulePath).FirstOrDefault(x => x.Contains("ModuleVersion"));
+            if (string.IsNullOrEmpty(versionLine))
+            {
+                return null;
+            }
+
+            var verRegex = new Regex(@"\d+\.\d+\.\d+");
+            if (!verRegex.IsMatch(versionLine))
+            {
+                return null;
+            }
+
+            var version = new TaskVersion(verRegex.Match(versionLine).Value)
+            {
+                IsTest = new Regex("(?i)(preview|test)").IsMatch(versionLine)
```

Comment:
>IsTest [](http://example.com/codeflow?start=16&length=6)

pretty sure TaskVersion already handles this?

Reply:
That's just a bool flag which can be used for comparison. No special handling needed.

Reply:
The Task Version could set it automatically if the version has suffix '-test' but not for '-peview'. So, again, pre-processing is needed.